crates/storage/provider/src/traits/executor.rs

```diff
@@ -45,6 +45,23 @@ pub trait BlockExecutor {
         senders: Option<Vec<Address>>,
     ) -> Result<(), BlockExecutionError>;
 
+    /// Runs the provided transactions and commits their state to the run-time database.
+    ///
+    /// The returned [BundleStateWithReceipts] can be used to persist the changes to disk, and
+    /// contains the changes made by each transaction.
+    ///
+    /// The changes in [BundleStateWithReceipts] have a transition ID associated with them: there is
+    /// one transition ID for each transaction (with the first executed tx having transition ID
+    /// 0, and so on).
+    ///
+    /// The second returned value represents the total gas used by this block of transactions.
+    fn execute_transactions(
```

Comment:
As a user it's a bit confusing now cuz there's 3 functions which execute right? And we don't return `BundleStateWithReceipts`, per the comment, it's just seemingly the Vec<Receipt> + transitionId